### ISSUE TITLE
Changing of cover art BREAKS the flac file?

### ISSUE THREAD

**Issue by gerchicov-bp** [taglib/taglib#668]
```
        FLAC::Picture* picture = new FLAC::Picture();
        TagLib::ByteVector bv = ByteVector((const char *)[albumArt bytes], [albumArt length]);
        picture->setData(bv);
        picture->setMimeType(String("image/png"));
        picture->setType(FLAC::Picture::FrontCover);
        
        FLAC::File *theFile = dynamic_cast<FLAC::File *>(_file.file());
        List<FLAC::Picture *> theList = theFile->pictureList();
        for (List<FLAC::Picture *>::Iterator it=theList.begin(); it != theList.end(); ++it)
        {
            FLAC::Picture *thePicture = *it;
            if (thePicture->type() == FLAC::Picture::FrontCover)
            {
                theFile->removePicture(thePicture);
                break;
            }
        }
//        theFile->removePictures();
        theFile->addPicture(picture);
        theFile->save();
```

With this code I'm trying to save the file as is was - without cover art tags but with cover art picture. But after the code above:
- vlc (windows) and Clementine (osx) can't play this file;
- vox (osx) can play it but the rewind and forward wind are disabled;
- vlc (osx) can play this file but it but it seems it has some serious troubles with caching by itself. I changed the coverart of the same file (about 3 pictures + removed picture) and replaced it multiple times but it shows random picture for a defined time (as if was cached);
- EasyTag shows info as if everything is done correctly.

Could anybody say where is my mistake?

**Comment by gerchicov-bp** [comment]
```
    if(block->code() == MetadataBlock::VorbisComment) {
      // Set the new Vorbis Comment block
      delete block;
      block = new UnknownMetadataBlock(MetadataBlock::VorbisComment, d->xiphCommentData);
      foundVorbisCommentBlock = true;
    }
```
what does this code actually do? It look likes a bug because file may have multiple tags and it seems this code replaces correct data with incorrect "proxy". But the main problem is that such resulting format is partially or fully rejected by many apps!

**Comment by TsudaKageyu** [comment]
The code replaces the old Vorbis comment block with a newly rendered one. It's correct and essential.

Note that the code block is preceded by this line where ```d->xiphCommentData``` is updated with significant data, not an incorrect proxy. 
```
  d->xiphCommentData = xiphComment()->render(false);
```

I think that ```FLAC::File``` has to fail to update tags all the time if you pointed out an actual bug.

**Comment by TsudaKageyu** [comment]
@gerchicov-bp, could you provide us with the file that you encountered the bug?
Please feel free to send an email to tsuda.kageyu@gmail.com directly.

**Comment by gerchicov-bp** [comment]
If flac file updated or not updated - in both cases it still should be at least playable on PC/Mac. But the code above breaks it. I forgot to say that I use taglib-objC:
https://github.com/eni9889/TagLib-ObjC
but it is just a wrapper and cover art methods are fully overridden by me.

I posted the resulting file here:
​http://www.filedropper.com/01-fergalicious111232_2

I have also found out that it works if I delete picture/pictures or add picture but never when I try to replace the picture (deleting + adding before the save() call).

**Comment by TsudaKageyu** [comment]
Hi @gerchicov-bp, thanks for providing the file. Please don't get me wrong, I didn't deny your report and the bug themselves.

**Comment by TsudaKageyu** [comment]
The seek table of the file is broken. I removed the seek table from the file manually, and it worked fine on my side. Please check out the file with your players:
https://drive.google.com/open?id=0B7nug4bMTx_rUFNHcHpWWjdCcTg

However, I couldn't find anything in TagLib which would alter a seek table... I'm afraid to say, could you check if your file was already broken?

**Comment by gerchicov-bp** [comment]
I've checked. Updated info:
- vox can play and scroll the file;
- vlc (both windows and osx, the latest 2.2.1) doesn't play it;
- vlc 2.0.4 (osx) could play the file I posted but I haven't a possibility to check it now with your updated file;
- mediamonkey (windows) can play and scroll the file;
- EasyTag shows that I was mistaken - my code fix this problem but adds another one: ```"artist"``` becomes ```"artist" - "artist"``` and then ```"artist" - "artist" - "artist"``` and the same is happened with other tags

Maybe it will be more helpful:
http://www.filedropper.com/01-fergalicious111232
it is not the original file but it was edited with one or few third party tag editors, it workable and after taglib processing it doesn't work and becomes the file I've already mentioned.

So the workable file:
http://www.filedropper.com/01-fergalicious111232
is transformed with taglib into the broken one:
http://www.filedropper.com/01-fergalicious111232_2

**Comment by TsudaKageyu** [comment]
Oh sorry. I'm going to check these files soon.

**Comment by TsudaKageyu** [comment]
I have one thing to confirm: Do you call ```save()``` method before the code block? Calling it twice may break a file as I reported at #622.

**Comment by gerchicov-bp** [comment]
Maybe. Maybe it doesn't depend on when to call ```save()```, just 2 times. But when I replaced double saving it seems to be workable.
But what if the user will call it twice? To close and open file?

**Comment by TsudaKageyu** [comment]
They are big bugs that (at least in my opinion) definitely needed to be fixed in the future. However, we found them recently, so we didn't have enough time to fix them before releasing v1.10BETA.

**Comment by FestusHagen** [comment]
#682 Closes this?

-Enjoy
